.github/workflows/ci-cd.yml

```diff
@@ -43,8 +43,21 @@ jobs:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
       
+      - name: Build versioned docker image
+        run: docker build --target production --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:$GITHUB_SHA .
+
       - name: Build prod docker image
         run: docker build --target production --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:prod .
       
-      - name: Push prod docker image to Docker Hub
+      - name: Push versioned image to Docker Hub
+        run: docker push ${{ secrets.DOCKER_USERNAME }}/todo-app:$GITHUB_SHA
+      
+      - name: Push prod image to Docker Hub
         run: docker push ${{ secrets.DOCKER_USERNAME }}/todo-app:prod
```

Comment:
Similarly, [Stack Overflow tells me](https://stackoverflow.com/questions/31963525/is-it-possible-for-image-to-have-multiple-tags) (i.e. I haven't tested!) that we should be able to combine these using the `--all-tags` flag